### PR TITLE
Replace gpu-to-vulkan pass with comp-to-vulkan pass

### PR DESCRIPTION
**Pull request by YangleiZouIntel**

### CODE REVIEW

**Review by earhart — DISMISSED** [review]
Awesome!

File: pmlc/conversion/gpu/passes.td [hunk 1]
```diff
@@ -17,4 +13,4 @@ def GpuKernelOutliningPass : Pass<"pmlc-kernel_outlining", "mlir::ModuleOp"> {
   ];
 }
 
-#endif // __PMLC_CONVERSION_GPU_TO_VULKAN_PASSES__
\ No newline at end of file
+#endif // __PMLC_GPU_KERNEL_OUTLINING_PASS__
```
Comment (earhart) [inline pmlc/conversion/gpu/passes.td:16]: It wouldn't be bad to put a newline in here at some point.

File: pmlc/conversion/comp_to_llvm/comp_to_vulkan.cc [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/IR/StandardTypes.h"
+#include "mlir/Support/LogicalResult.h"
+#include "mlir/Transforms/DialectConversion.h"
+
+#include "pmlc/conversion/comp_to_llvm/pass_detail.h"
+#include "pmlc/conversion/comp_to_llvm/utils.h"
+#include "pmlc/dialect/comp/ir/dialect.h"
+
+namespace pmlc::conversion::comp_to_llvm {
```
Comment (earhart) [inline pmlc/conversion/comp_to_llvm/comp_to_vulkan.cc:23]: I wonder whether we ought to have a separate namespace here?

**Comment by YangleiZouIntel** [comment]
> Awesome!

Thanks, I will fix the items you mentioned. This is mainly built on the previous pass, rewriting comp::schedule_func and init/deinit op. It passed resnet50 on my local machine but somehow failed on CI, I will take a look. 

We actually have a more aggreesive version which  build on the same comp IR with ocl backend, conveting  comp.alloc/dealloc comp.schedule_read/write, but it currently does not passing all backend tests. So I think I will make that a separate PR and make sure resnet50 and backend tests passed on each patch/PR.

File: pmlc/conversion/comp_to_llvm/comp_to_vulkan.cc (hunk 2 above)
Comment (flaub) [reply pmlc/conversion/comp_to_llvm/comp_to_vulkan.cc:23]: I think this is correct, it should follow the directory structure, and `comp_to_vulkan` lives in `comp_to_llvm`

File: pmlc/rt/vulkan/wrappers.cc [hunk 3]
```diff
@@ -113,17 +119,17 @@ struct Registration {
...
+    registerSymbol("vkDeinit", reinterpret_cast<void *>(vkDeinit));
+    registerSymbol("vkCreateLaunchKernelAction",
+                   reinterpret_cast<void *>(vkCreateLaunchKernelAction));
+    registerSymbol("vkCreateMemoryTransferAction",
+                   reinterpret_cast<void *>(vkCreateMemoryTransferAction));
+    registerSymbol("vkSetLaunchKernelAction",
+                   reinterpret_cast<void *>(vkSetLaunchKernelAction));
+    registerSymbol("vkRun", reinterpret_cast<void *>(vkRun));
+    registerSymbol("VkWait", reinterpret_cast<void *>(VkWait));
+    registerSymbol("VkScheduleFunc", reinterpret_cast<void *>(VkScheduleFunc));
```
Comment (flaub) [inline pmlc/rt/vulkan/wrappers.cc:132]: We should try to make the runtime functions consistent, I notice that some are `VkXXX` and others are `vkXXX`

**Review by flaub — APPROVED** [review]

**Pull request merged**